packages/smooth_app/lib/pages/user_management/sign_up_page.dart

```diff
@@ -0,0 +1,311 @@
+  final TextEditingController _password1Controller = TextEditingController();
+  final TextEditingController _password2Controller = TextEditingController();
+  final TextEditingController _brandController = TextEditingController();
+
+  bool _foodProducer = false;
+  bool _agree = false;
+  bool _subscribe = false;
+  bool _disagreed = false;
+  late bool _popEd;
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    final Size size = MediaQuery.of(context).size;
+
+    // TODO(monsieurtanuki): translations
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Sign Up'),
+        backgroundColor: Colors.transparent,
+        elevation: 0,
+        iconTheme: IconThemeData(
+          color: Theme.of(context).colorScheme.primary,
+        ),
+      ),
+      body: Form(
+        onChanged: () {
+          setState(() {});
+        },
+        key: _formKey,
+        child: ListView(
+          padding: EdgeInsets.symmetric(horizontal: size.width * 0.05),
+          children: <Widget>[
+            SmoothTextFormField(
+              textInputType: TextInputType.name,
+              type: TextFieldTypes.PLAIN_TEXT,
+              controller: _displayNameController,
+              hintText: 'Display Name',
+              prefixIcon: const Icon(Icons.person),
+              autofillHints: const <String>[AutofillHints.name],
+              validator: (String? value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter the display name';
+                }
+                return null;
+              },
+            ),
+            const SizedBox(height: space),
+            SmoothTextFormField(
+              textInputType: TextInputType.emailAddress,
+              type: TextFieldTypes.PLAIN_TEXT,
+              controller: _emailController,
+              hintText: 'E-mail',
+              prefixIcon: const Icon(Icons.person),
+              autofillHints: const <String>[AutofillHints.email],
+              validator: (String? value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter the e-mail';
+                }
+                if (!UserManagementHelper.isEmailValid(value)) {
+                  return 'Please enter a valid email';
+                }
+                return null;
+              },
+            ),
+            const SizedBox(height: space),
+            SmoothTextFormField(
+              type: TextFieldTypes.PLAIN_TEXT,
+              controller: _userController,
+              hintText: 'Username',
+              prefixIcon: const Icon(Icons.person),
+              autofillHints: const <String>[AutofillHints.username],
+              validator: (String? value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter a username';
```

Comment:
I realise we probably have no way to check whether it's available (or scrape the response ?)

Reply:
actually:
https://world.openfoodfacts.org/contributor/teolemon.json returns a JSON response
https://world.openfoodfacts.org/contributor/teolemonfake.json returns an HTML page with Unknown User

Reply:
Anyway that's one of the few error message we can get with `register`: "user already exists" (always in english, though).